File: facebook/junitTests/src/test/java/com/facebook/FacebookTestCase.java
/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * You are hereby granted a non-exclusive, worldwide, royalty-free license to use,
 * copy, modify, and distribute this software in source code or binary form for use
 * in connection with the web services and APIs provided by Facebook.
 *
 * As with any software that integrates with the Facebook platform, your use of
 * this software is subject to the Facebook Developer Principles and Policies
 * [http://developers.facebook.com/policy/]. This copyright notice shall be
 * included in all copies or substantial portions of the software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
 * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
 * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
 * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

package com.facebook;

import org.junit.Before;
import org.junit.runner.RunWith;
import org.mockito.MockitoAnnotations;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;
import org.robolectric.shadows.ShadowLog;

// ShadowLog is used to redirect the android.util.Log calls to System.out
@Config(shadows = {ShadowLog.class})
@RunWith(RobolectricTestRunner.class)

/**
 * Base class for Robolectric tests.
 * Important: the classes that derive from this should end with Test (i.e. not Tests) otherwise the
 * gradle task "test" doesn't pick them up.
 */
public abstract class FacebookTestCase {
    @Before
    public void setUp() {
        ShadowLog.stream = System.out;
        MockitoAnnotations.initMocks(this);
    }
}

File: facebook-testutil/src/main/java/com/facebook/FacebookTestCase.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook

import org.junit.Before
import org.junit.runner.RunWith
import org.mockito.MockitoAnnotations
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config
import org.robolectric.shadows.ShadowLog

// ShadowLog is used to redirect the android.util.Log calls to System.out
@Config(shadows = [ShadowLog::class], sdk = [21])
@RunWith(RobolectricTestRunner::class)
/**
 * Base class for Robolectric tests. Important: the classes that derive from this should end with
 * Test (i.e. not Tests) otherwise the gradle task "test" doesn't pick them up.
 */
abstract class FacebookTestCase {
  @Before
  open fun setUp() {
    ShadowLog.stream = System.out
    MockitoAnnotations.initMocks(this)
  }
}
